src/statsd_client.cpp

```diff
@@ -50,14 +50,18 @@ inline bool fequal(float a, float b)
     return ( fabs(a - b) < epsilon );
 }
 
+FastRandomContext insecure_rand;
```

Comment:
If you use the `thread_local` keyword here, you remove the need for `rnd_cs`. Better to have as less locking/blocking as possible in metric methods.